### PR TITLE
Verify dependency versions for release packages

### PR DESCRIPTION
**Pull request by pakrym**
So we don't ship release packages with pre-release dependencies.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs.ServiceFabricProcessor/src/Microsoft.Azure.EventHubs.ServiceFabricProcessor.csproj
Comment (pakrym) [inline sdk/eventhub/Microsoft.Azure.EventHubs.ServiceFabricProcessor/src/Microsoft.Azure.EventHubs.ServiceFabricProcessor.csproj:4]: ..\..\Microsoft.Azure.EventHubs\src\Microsoft.Azure.EventHubs.csproj dependency is in beta

**Review by tg-msft — APPROVED** [review]
Nice!

**Review by AlexGhiondea — APPROVED** [review]

**Review by jsquire — APPROVED** [review]

**Comment by jsquire** [comment]
//fyi:  @JamesBirdsall

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets:179]: Doesn't nuget already do this for us? https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5104

It looks like they made that a warning (https://github.com/NuGet/NuGet.Client/pull/1073) but given we have warnings-as-errors I would expect this to already fail the build. If that isn't the case we should see if we can figure out why.

Comment (pakrym) [reply eng/Directory.Build.Data.targets:179]: It didn't catch the issue we had in one of the projects. I'll take a look at why

Comment (pakrym) [reply eng/Directory.Build.Data.targets:179]: I think it's because there is always an alpha package version when working locally so nu5104 might kick in during a release build.

Comment (pakrym) [reply eng/Directory.Build.Data.targets:179]: Yep.

```
 dotnet pack /p:SkipDevBuildNumber=true
Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored D:\github\azure\net2\sdk\eventhub\Microsoft.Azure.EventHubs\src\Microsoft.Azure.EventHubs.csproj (in 562 ms).
  Restored D:\github\azure\net2\sdk\eventhub\Microsoft.Azure.EventHubs.ServiceFabricProcessor\src\Microsoft.Azure.EventHubs.ServiceFabricProcessor.csproj (in 562 ms).
  Ran ApiCompat against C:\Users\pakrym\.nuget\packages\\microsoft.azure.eventhubs\4.3.1\lib\netstandard2.0\Microsoft.Azure.EventHubs.dll using assemblies from D:\github\azure\net2\artifacts\obj\Microsoft.Azure.EventHubs\Debug\netstandard2.0\
  Microsoft.Azure.EventHubs -> D:\github\azure\net2\artifacts\bin\Microsoft.Azure.EventHubs\Debug\netstandard2.0\Microsoft.Azure.EventHubs.dll
  Microsoft.Azure.EventHubs.ServiceFabricProcessor -> D:\github\azure\net2\artifacts\bin\Microsoft.Azure.EventHubs.ServiceFabricProcessor\Debug\netstandard2.0\Microsoft.Azure.EventHubs.ServiceFabricProcessor.dll
C:\Program Files\dotnet\sdk\3.1.403\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(198,5): error NU5104: A stable release of a package should not have a prerelease dependency. Either modify the version spec of dependency "Microsoft.Azure.EventHubs [4.4.0-beta.1, )" or update the version field in the nuspec. [D:\github\azure\net2\sdk\eventhub\Microsoft.Azure.EventHubs.ServiceFabricProcessor\src\Microsoft.Azure.EventHubs.ServiceFabricProcessor.csproj]
  Successfully created package 'D:\github\azure\net2\artifacts\packages\Debug\/Microsoft.Azure.EventHubs.ServiceFabricProcessor\Microsoft.Azure.EventHubs.ServiceFabricProcessor.0.5.6.symbols.nupkg'.
```

Considering this what would you prefer me to do?

Comment (weshaggard) [reply eng/Directory.Build.Data.targets:179]: I guess we simply need to figure out how important it is for us to have this issue show up in PR validation vs showing up only when we do release builds. If we are OK with having it wait until release builds then I'd definitely remove this extra layer of checks so we don't have to try and maintain it. I'll leave that up to you guys.

Comment (weshaggard) [reply eng/Directory.Build.Data.targets:179]: If you want this earlier warning you could also consider depending on the Nuget warning but having a wrapper target that disables the dev versioning for just that invocation.

**Pull request merged**
